### DISCUSSION TITLE
x-axis on State Transitions panel does not align with Plot panel

### DISCUSSION THREAD

**Discussion by rgov** [foxglove/community#530] [Q&A]
Foxglove 1.8.0, macOS 12.3

I have a Plot panel and State Transitions panel aligned vertically with one another. Both use the `header.stamp` for their *x* axes. But the data points do not align between the two panels, because the width of the plots are slightly different, probably due to the absence of the *y* axis labels on the State panel.

The peak on the Depth plot should align exactly with the end of the Downcast state, but they are slightly offset.

<img width="725" alt="image" src="https://user-images.githubusercontent.com/108767/164914879-62d77ae1-f359-4ad8-a0fb-e4fb35c63ec1.png">

It's more exaggerated at other parts of the plot:

<img width="730" alt="image" src="https://user-images.githubusercontent.com/108767/164914889-f8a1127a-32e4-42b9-9a88-65952525f30b.png">

**Comment by defunctzombie** [comment]
Summarizing our discussion we had in person about this. The challenge here is that chart.js computes these tick labels automatically and since these are plots in two separate panels we'd need to have away to synchronize the gutter.

While we do have a way to synchronize the hover values (because thats a common part of the app), we don't have anything specific to gutters or other bespoke elements of panels.

Another idea I mentioned was to provide a setting to set a fixed gutter size. I've not found any chartjs configuration (https://www.chartjs.org/docs/latest/axes/labelling.html) that lets us set this fixed size - only a padding around the text. So the next idea could be some sort of custom tick formatter where a user could enter a setting of `####.##` and we'd use that format for the y-axis ticks and pad out the leading numbers.

I agree there is something visually appealing to lining up axis for stacked plot layouts.
